terraform/main.tf

```diff
@@ -50,6 +50,44 @@ resource "azurerm_resource_group" "grp" {
   location = var.resource_group_location
 }
 
+# Keyvault for Saving Secrets
+resource "azurerm_key_vault" "key_vault" {
+  name                       = "${var.project_name}-keyvault"
+  location                   = azurerm_resource_group.grp.location
+  resource_group_name        = azurerm_resource_group.grp.name
+  tenant_id                  = data.azurerm_client_config.current.tenant_id
+  sku_name                   = "standard"
+  soft_delete_retention_days = 7
+
+  access_policy {
+    tenant_id = data.azurerm_client_config.current.tenant_id
+    object_id = data.azurerm_client_config.current.object_id
+
+    secret_permissions = [
+      "Set",
+      "Get",
+      "List",
+      "Delete",
+      "Purge",
+      "Recover"
+    ]
+  }
+}
+
+# Create Secret for Admin Username
+resource "azurerm_key_vault_secret" "admin_username" {
+  name         = "admin-username"
```

Comment:
This key vault secret should be named `POSTGRES-LOGIN-NAME` to match what the applications expect as seen [here](https://github.com/Azure-Samples/acme-fitness-store#create-azure-key-vault-and-store-secrets)